packages/vscode-extension/src/devices/preview.ts

```diff
@@ -33,10 +33,15 @@ export class Preview implements Disposable {
         reject(new Error("Preview server exited without URL"));
       });
 
+      const streamURLRegex = /(http:\/\/[^ ]*stream\.mjpeg)/;
+
       lineReader(subprocess).onLineRead((line) => {
-        if (line.includes("http://")) {
-          Logger.debug(`Preview server ready ${line}`);
-          this.streamURL = line;
+        const match = line.match(streamURLRegex);
```

Comment:
Suggestion: you can do: `const [_full, url] = line.match(streamURLRegex);`

Reply:
Also, you can simplify the regex to `/http:\/\/.+stream\.mjpeg/;` and just use full match.

Reply:
I can't do that because match maybe null, I could do deconstruction inside the if(match) scope, but it seems unnecessary.